File: src/Services/Roslyn/Markify.Services.Roslyn.Fixtures/Projects/Source/Members/Events.cs
#region Class

using System;

namespace Class
{
    public class TypeWithNoEvent
    {
    }

    public abstract class AbstractTypeWithEvents
    {
        public abstract event EventHandler AbstractEvent;
    }

    public class TypeWithEvents : AbstractTypeWithEvents
    {
        event EventHandler Event;

        private event EventHandler PrivateEvent;

        internal event EventHandler InternalEvent;

        protected internal event EventHandler ProtectedInternalEvent;

        protected event EventHandler ProtectedEvent;

        public static event EventHandler StaticEvent;

        public virtual event EventHandler VirtualEvent;

        public sealed override event EventHandler SealedEvent;

        public event EventHandler<EventArgs> GenericEvent;

        public event EventHandler ExplicitEvent
        {
            add { }
            remove { }
        }
    }
}

#endregion

#region Interface

namespace Interface
{
    public interface TypeWithNoEvent
    {
    }

    public interface TypeWithEvents
    {
        event EventHandler Event;

        event EventHandler<EventArgs> GenericEvent;
    }
}

#endregion

#region Struct

namespace Struct
{
    public struct TypeWithNoEvent
    {
    }

    public struct TypeWithEvents
    {
        event EventHandler Event;

        private event EventHandler PrivateEvent;

        internal event EventHandler InternalEvent;

        public static event EventHandler StaticEvent;

        public event EventHandler<EventArgs> GenericEvent;

        public event EventHandler ExplicitEvent
        {
            add { }
            remove { }
        }
    }
}

#endregion

File: src/Services/Roslyn/Markify.Services.Roslyn.Fixtures/Projects/Source/Members/Events.vb
#Region "Class"

Namespace Class
    Public Class TypeWithNoEvent
    End Class

    Public MustInherit Class AbstractTypeWithEvents
        Public MustInherit Event AbstractEvent As EventHandler
    End Class

    Public Class TypeWithEvents
        Inherits AbstractTypeWithEvents

        Event Event As EventHandler

        Private PrivateEvent As EventHandler

        Friend Event InternalEvent As EventHandler

        Protected Friend Event ProtectedInternalEvent As EventHandler

        Protected Event ProtectedEvent As EventHandler

        Public Shared Event StaticEvent As EventHandler

        Public Overridable Event VirtualEvent As EventHandler

        Public NotOverridable Overrides Event SealedEvent As EventHandler

        Public Event GenericEvent As EventHandler(Of EventArgs)

        Public Custom Event ExplicitEvent As EventHandler
            AddHandler(ByVal value As EventHandler)
            End AddHandler
            RemoveHandler(ByVal value As EventHandler)
            End RemoveHandler
            RaiseEvent(ByVal sender As Object, ByVal e As EventArgs)
            End RaiseEvent
        End Event
    End Class
End Namespace

#End Region

#Region "Interface"

Namespace Interface
    Public Interface TypeWithNoEvent
    End Interface

    Public Interface TypeWithEvents
        Event Event As EventHandler

        Event GenericEvent As EventHandler(Of EventArgs)
    End Interface
End Namespace

#End Region

#Region "Struct"

Namespace Struct
    Public Structure TypeWithNoEvent
    End Structure

    Public Structure TypeWithEvents
        Event Event As EventHandler

        Private Event PrivateEvent As EventHandler

        Friend Event InternalEvent As EventHandler

        Public Shared Event StaticEvent As EventHandler

        Public Event GenericEvent As EventHandler(Of EventArgs)

        Public Custom Event ExplicitEvent As EventHandler
            AddHandler(ByVal value As EventHandler)
            End AddHandler
            RemoveHandler(ByVal value As EventHandler)
            End RemoveHandler
            RaiseEvent(ByVal sender As Object, ByVal e As EventArgs)
            End RaiseEvent
        End Event
    End Structure
End Namespace

#End Region